lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -0,0 +1,651 @@
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'field_errors': field_errors
+                }
+            )
+
+        try:
+            coach = User.objects.get(email=valid_input['coach_email'])
+        except User.DoesNotExist:
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={
+                    'error_code': 'coach_user_does_not_exist'
+                }
+            )
+
+        # TODO: verify that the transaction.atomic is enough for EDX
+        with transaction.atomic():
+            ccx_course_object = CustomCourseForEdX(
+                course_id=master_course_object.id,
+                coach=coach,
+                display_name=valid_input['display_name'])
+            ccx_course_object.save()
+
+            # Make sure start/due are overridden for entire course
+            start = TODAY().replace(tzinfo=pytz.UTC)
+            override_field_for_ccx(ccx_course_object, master_course_object, 'start', start)
+            override_field_for_ccx(ccx_course_object, master_course_object, 'due', None)
+
+            # Enforce a static limit for the maximum amount of students that can be enrolled
+            override_field_for_ccx(
+                ccx_course_object,
+                master_course_object,
+                'max_student_enrollments_allowed',
+                valid_input['max_students_allowed']
+            )
+
+            # Hide anything that can show up in the schedule
+            hidden = 'visible_to_staff_only'
+            for chapter in master_course_object.get_children():
+                override_field_for_ccx(ccx_course_object, chapter, hidden, True)
+                for sequential in chapter.get_children():
+                    override_field_for_ccx(ccx_course_object, sequential, hidden, True)
+                    for vertical in sequential.get_children():
+                        override_field_for_ccx(ccx_course_object, vertical, hidden, True)
+
+            # make the coach user a coach on the master course
+            make_user_coach(coach, master_course_key)
+
+            # pull the ccx course key
+            ccx_course_key = CCXLocator.from_course_locator(master_course_object.id, ccx_course_object.id)
+            # enroll the coach in the newly created ccx
+            email_params = get_email_params(
+                master_course_object,
+                auto_enroll=True,
+                course_key=ccx_course_key,
+                display_name=ccx_course_object.display_name
+            )
+            enroll_email(
+                course_id=ccx_course_key,
+                student_email=coach.email,
+                auto_enroll=True,
+                email_students=True,
+                email_params=email_params,
+            )
+            # assign coach role for the coach to the newly created ccx
+            assign_coach_role_to_ccx(ccx_course_key, coach, master_course_object.id)
```

Comment:
I talked with other people here and I think we are OK with the kind of errors we handle, so I will leave it as it is